cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -991,6 +991,9 @@ TEST(GroupBy, SumOnlyStringAndDictKeys) {
                                            {
                                                {"hash_sum", nullptr},
                                            }));
+    if (key_type->Equals(utf8())) {
```

Comment:
Why does it only require sorting for string keys?

Reply:
SortBy doesn't work for string dictionary type. And accidentally the order does not change for this test case, so it works across the systems. If you can tell me how to sort dictionary array I will change it.

Reply:
I'll push a change to support sorting by dictionary arrays